### PR TITLE
Test GCE PD unmounts and detaches when the namespace of the pvc&pod is deleted.

### PR DESCRIPTION
**Pull request by copejon**
Addition to Persistent Volume E2E testing.  On a GCE cluster, create a pv, pvc, and client pod.  Delete the namespace and check that the disk detaches successfully.
@jeffvance

~~DEPENDENT ON~~ #34353 merged.  No dependencies.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35831)

### CODE REVIEW

**Comment by jeffvance** [comment]
LGTM. Thanks for this important refactoring Jon!

**Comment by copejon** [comment]
cc @erinboyd

**Comment by copejon** [comment]
cc @jsafrane

**Comment by jsafrane** [comment]
@k8s-bot ok to test

**Comment by jsafrane** [comment]
preliminary LGTM, waiting for #34353 to get merged

**Comment by jeffvance** [comment]
@copejon is off for the holiday break. Is there a way to get extra time on this pr?

----- Original Message -----

**Comment by copejon** [comment]
Looks like it's failing while waiting for the namespace to delete.  This hasn't cropped up in my own testing.  Thinking about it, it's not really necessary to wait for the ns to delete given that it is immediately followed by waiting for the PD to detach.

**Comment by copejon** [comment]
So it passes, but the question remains: why does it timeout waiting for the namespace?  Does the CI need more than 1 minute to perform the cleanup?

**Comment by jsafrane** [comment]
@copejon, when a namespace is deleted, all pods in it are deleted and it can take quite a long time as docker tries to kill them in a nice way (sigterm) and only after a timeout it kills them.

**Comment by copejon** [comment]
Oi... rebase screw up.  Fixing now.

File: pkg/generated/bindata.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ func (fi bindataFileInfo) Sys() interface{} {
...
...56\x6b\xca\x5b\x59\x84\xc7\xf2\x1f\xde\x86\x1c\xc1\x99\xad\x8d\xec\xdc\x7c\x89\x5b\xe8\x22\x5a\x38\xfb\x89\x59\x38\x95\xf8\x9e\x9c\x67\x76\x06\x05\x85\x40\x17\x01\x0b\x8b\x74\x21\xca\x4a\x93\xc7\x35\x69\x6d\xe3\x13\x6a\x56\x80\x73\x5f\x28\x89\xe3\xba\xf0\xb8\xb4\x19\xc4\x8b\xb7\x4b\x9c\x38\xda\x9e\x04\xbe\xe6\xd3\xc9\xa0\x97\xa4\x7d\x4c\x7b\xfc\x70\x98\x25\xfd\xa7\x09\x1b\x13\xf1\x84\xce\x95\xff\x8f\x77\xb8\xe5\xf5\xa1\x97\x66\xcf\x0f\x30\x19\x30\x6f\x26\x7c\xc0\xa5\x13\xc6\x6b\x11\xac\xcb\x60\xec\xc8\x48\x61\x60\x5c\x3b\x6e\xe7\xc5\xaa\x4d\xbb\xb2\xbb\x6a\x06\x5e\x15\xa5\x50\xba\x9b\xdb\xf2\x65\x3c\x9f\xce\x8f\xae\x5a\x49\xbb\x49\x3c\xb1\x26\x90\x61\xbd\xcb\x8a\x0b\x36\x9d\x3d\xab\xb4\x50\x66\x08\xf9\x5a\x38\x4f\x61\x74\xba\x3c\xc6\xc1\x15\xaf\xa9\x7b\x46\x0e\x8f\x4c\x6e\xa5\x32\x45\x06\x83\x95\x0a\xf1\x07\x7c\x43\x86\x5c\xbb\xa0\x85\x25\xa9\x02\xeb\x1f\x76\xd3\x84\x3f\xb5\x39\xbe\xdb\xee\xf0\xa4\xd5\xcd\xa0\x15\x9e\x09\x53\xd4\xa2\x20\x5c\x92\x28\x9b\xed\xd2\xb5\x13\x1a\x8f\xad\x2b\x7d\x06\xa6\xda\xa6\x7e\xd4\x1b\x42\x1b\x8e\xf6\x0d\x3c\x1c\x41\xfa\x64\xb8\x9b\x9a\x01\x99\xf8\x5a\xff\xc7\x0d\x5d\x1a\xd8\x08\x7f\x7b\xb7\xf2\x77\x00\x00\x00\xff\xff\xaa\xa3\x15\xc3\x6a\x04\x00\x00")
```
Comment (jsafrane) [inline pkg/generated/bindata.go]: these changes look suspicious. Are they part of this PR?

Comment (copejon) [inline pkg/generated/bindata.go]: No... That's very odd.  Will remove.

Comment (copejon) [inline pkg/generated/bindata.go]: @jsafrane Looks like it's introduced when using the `$KPATH/hooks/pre-commit` hook.  I did a `git checkout UPSTREAM/master pkg/generated/bindata.go` and then a `commit --amend`.  After the pre-commit script finished, the file is listed as `modified`.

**Comment by jsafrane** [comment]
@copejon, please rebase your code so we can merge it.

**Comment by copejon** [comment]
@jsafrane thanks, rebased.

**Comment by jsafrane** [comment]
@k8s-bot kops aws e2e test this

**Comment by copejon** [comment]
@ixdy can you PTAL / approve?

**Comment by jeffvance** [comment]
@copejon my pr #40530  does significant restructuring to _persistent_volumes.go_. Let me know if you want any help with the merge.

**Comment by jsafrane** [comment]
@k8s-bot kops aws e2e test this
@k8s-bot cri e2e test this

**Pull request merged**